website/docs/guides/configuring-oidc-with-keycloak.mdx

```diff
@@ -0,0 +1,272 @@
+
+1. Navigate to the `Credentials` tab in the top menu
+2. Click on the `Set password` button
+
+![Keycloak create user step 4](img/kc_step_21.png)
+
+In the dialog that opens up, set a password for your user, confirm it, optionally mark it as not-temporary and click on `Save`.
+
+![Keycloak create user step 5](img/kc_step_22.png)
+
+You should now see a password configured for your user.
+
+![Keycloak create user step 6](img/kc_step_23.png)
+
+Now we need to assign the `wego-admin` role we created earlier to our user. To do that:
+
+1. Navigate to the `Role mapping` tab in the top menu
+2. Click on the `Assign role` button
+
+![Keycloak create user step 7](img/kc_step_24.png)
+
+In the dialog that opens up:
+
+1. Select the `Filter by clients` option in the dropdown
+2. Type in your client name in the search bar
+3. Tick the checkbox next to the role
+4. Click on `Assign`
+
+![Keycloak create user step 8](img/kc_step_25.png)
+
+The role should now appear in the `Role mapping` section and our user should be fully configured.
+
+## Configuring Weave GitOps to use our new Keycloak Client
+
+Now that our Keycloak configuration is done, it's time to link our Weave GitOps deployment to it.
+
+### Creating the `oidc-auth` secret
+
+To configure Weave GitOps for OIDC authentication via Keycloak, we need to configure the `oidc-auth` secret. We need to set:
+
+- the `issuerURL` to the URL of our keycloak instance, followed by `/realms/<realm_name>`
+- the `redirectURL` to the URL of our Weave GitOps instance, followed by `/oauth2/callback`
+- the `clientID` to the id of the client we created in the previous steps
+- the `clientSecret` to the secret we copied a few steps ago
+- the `customScopes` to `email`
+- the `claimGroups` to `groups` to work with our mapper, mapping from roles->groups
+- the `claimUsername` to `sub`
+
+<details><summary>Expand to see secret definition </summary>
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+    name: oidc-auth
+stringData:
+    issuerURL: https://auth.mydomain.com/keycloak/realms/demo
+    redirectURL: https://gitops.mydomain.com/oauth2/callback
+    clientID: weave-gitops
+    clientSecret: N8jDkMdghg38jiw52VHeTH1V7WUmM1tv
+    customScopes: email
+    claimGroups: groups
+    claimUsername: sub
+```
+
+</details>
+
+After this secret is created, you may need to delete the Weave GitOps pods in order to restart the app and load the new config.
+
+### Setting up RBAC
+
+Once Weave GitOps is configured for OIDC, we need a way to map permissions to the groups. To do that, we need to create role bindings for our `wego-admin` group.
+The following example assumes that the ClusterRole `wego-admin-cluster-role` and the namespaced Role `wego-admin-role` already exist. See the
+[recommendations on setting up RBAC](../../enterprise/getting-started/install-enterprise/#recommended-rbac-configuration) for details.
+
+<details><summary>Expand to see group role bindings</summary>
```

Comment:
I'm not sure I understand why in this example the user would need to create a namespaced and a cluster-scoped binding. The ClusterRoleBinding should be sufficient in this case.

Reply:
I believe it was a bad copy-paste job from here: https://github.com/weaveworks/weave-gitops/blob/main/website/versioned_docs/version-0.36.0/enterprise/getting-started/install-enterprise.mdx#L841-L869
Or rather, the configuration I ended up with after tinkering with that example for a while and I forgot to proofread it 😅